tools/amd_build/pyHIPIFY/hipify-python.py

```diff
@@ -0,0 +1,946 @@
+    for kernel in get_kernel_positions:
+        # Get kernel components
+        params = grab_method_and_template(kernel)
+
+        # Find paranthesis after kernel launch
+        paranthesis = string.find("(", kernel["end"])
+
+        # Extract cuda kernel
+        cuda_kernel = string[params[0]["start"]:paranthesis+1]
+
+        # Keep number of kernel launch params consistent (grid dims, group dims, stream, dynamic shared size)
+        num_klp = len(extract_arguments(0, kernel["group"].replace("<<<", "(").replace(">>>", ")")))
+
+        # Transform cuda kernel to hip kernel
+        hip_kernel = "hipLaunchKernelGGL(" + cuda_kernel[0:-1].replace(">>>", ", 0"*(4-num_klp) + ">>>").replace("<<<", ", ").replace(">>>", ", ")
+
+        # Replace cuda kernel with hip kernel
+        output_string = output_string.replace(cuda_kernel, hip_kernel)
+
+        # Update the statistics
+        stats["kernel_launches"].append(hip_kernel)
+
+    return output_string
+
+
+def find_paranthesis_end(input_string, start):
+    inside_paranthesis = False
+    parans = 0
+    pos = start
+    p_start, p_end = -1, -1
+
+    while pos < len(input_string):
+        if input_string[pos] == "(":
+            if inside_paranthesis is False:
+                inside_paranthesis = True
+                parans = 1
+                p_start = pos
+            else:
+                parans += 1
+        elif input_string[pos] == ")" and inside_paranthesis:
+            parans -= 1
+
+            if parans == 0:
+                p_end = pos
+                return p_start, p_end
+
+        pos += 1
+    return None, None
+
+
+def disable_asserts(input_string):
+    """ Disables regular assert statements
+    e.g. "assert(....)" -> "/*assert(....)*/"
+    """
+    output_string = input_string
+    asserts = list(re.finditer(r"\bassert[ ]*\(", input_string))
+    for assert_item in asserts:
+        p_start, p_end = find_paranthesis_end(input_string, assert_item.end()-1)
+        start = assert_item.start()
+        output_string = output_string.replace(input_string[start:p_end+1], "")
+    return output_string
+
+
+def disable_function(input_string, function, replace_style):
+    """ Finds and disables a function in a particular file.
+
+    If type(function) == List
+        function - The signature of the function to disable.
+            e.g. ["bool", "overlappingIndices", "(const Tensor& t)"]
+            disables function -> "bool overlappingIndices(const Tensor& t)"
+
+    If type(function) == String
+        function - Disables the function by name only.
+            e.g. "overlappingIndices"
+
+    replace_style - The style to use when stubbing functions.
+        0 - Remove the function entirely (includes the signature).
+        1 - Stub the function and return an empty object based off the type.
+        2 - Add !defined(__HIP_PLATFORM_HCC__) preprocessors around the function.
+        3 - Add !defined(__HIP_DEVICE_COMPILE__) preprocessors around the function.
```

Comment:
What's the difference between these two preprocessors? How do I decide which one?

Reply:
`__HIP_PLATFORM_HCC__` is turned on if the compiler is `hcc`
`__HIP_DEVICE_COMPILE__` is turned on in the device path when using either `hcc` or `nvcc` through `HIP`

see: https://github.com/ROCm-Developer-Tools/HIP/blob/master/include/hip/hip_common.h

Reply:
Put it in the code? :)

Reply:
commented in enum